java/src/Ice/src/main/java/com/zeroc/IceInternal/RoutableReference.java

```diff
@@ -361,13 +378,37 @@ public boolean equals(java.lang.Object obj) {
   }
 
   @Override
-  public RequestHandler getRequestHandler(com.zeroc.Ice._ObjectPrxI proxy) {
-    return getInstance().requestHandlerFactory().getRequestHandler(this, proxy);
+  RequestHandler getRequestHandler() {
+    com.zeroc.IceInternal.Instance instance = getInstance();
+    if (_collocationOptimized) {
+      com.zeroc.Ice.ObjectAdapter adapter = instance.objectAdapterFactory().findObjectAdapter(this);
+      if (adapter != null) {
+        return new CollocatedRequestHandler(this, adapter);
+      }
+    }
+
+    var handler = new ConnectRequestHandler(this);
+    if (instance.queueRequests()) {
+      final ConnectRequestHandler h = handler;
+      instance
+          .getQueueExecutor()
+          .executeNoThrow(
+              new Callable<Void>() {
+                @Override
+                public Void call() {
+                  getConnection(h);
+                  return null;
+                }
```

Comment:
Is the Callable required here? can't we just pass a lambda like in:

```
instance.getQueueExecutor().executeNoThrow(() -> {
            getConnection(h);
            return null;
        });
```

Reply:
This was what we were doing before, and I just copied it over.
But you're right, I replaced this with a lambda instead.